src/coreclr/nativeaot/Runtime/gcrhenv.cpp

```diff
@@ -1181,10 +1181,15 @@ bool GCToEEInterface::EagerFinalized(Object* obj)
     // Managed code should not be running.
     ASSERT(GCHeapUtilities::GetGCHeap()->IsGCInProgressHelper());
 
+    // the lowermost 1 bit is reserved for storing additional info about the handle
+    const uintptr_t HandleTagBits = 1;
+
     WeakReference* weakRefObj = (WeakReference*)obj;
-    OBJECTHANDLE handle = (OBJECTHANDLE)(weakRefObj->m_HandleAndKind & ~(uintptr_t)1);
-    HandleType handleType = (weakRefObj->m_HandleAndKind & 1) ? HandleType::HNDTYPE_WEAK_LONG : HandleType::HNDTYPE_WEAK_SHORT;
-    weakRefObj->m_HandleAndKind &= (uintptr_t)1;
+    OBJECTHANDLE handle = (OBJECTHANDLE)(weakRefObj->m_taggedHandle & ~HandleTagBits);
+    _ASSERTE((weakRefObj->m_taggedHandle & 2) == 0);
+    HandleType handleType = (weakRefObj->m_taggedHandle & 1) ? HandleType::HNDTYPE_WEAK_LONG : HandleType::HNDTYPE_WEAK_SHORT;
+    // keep the bit that indicates whether this reference was tracking resurrection, clear the rest.
+    weakRefObj->m_taggedHandle &= (uintptr_t)1;
```

Comment:
The correct value is actually `(uintptr_t)1`, since that is the tag which we are preserving. 
It happens to match `HandleTagBits`, but only because NativeAOT does not yet support weak COM references.

Reply:
I guess this will have to be fixed or folded with the CoreCLR version anyways when weak COM references are implemented, so we could use HandleTagBits.